### PR TITLE
Update ineligible_endpoints.yaml

### LINKED ISSUE #109379: Eligibility for conformance: Node endpoints: read/replaceCoreV1NodeStatus
In the K8s Conformance meeting of 6 April the testing of Node endpoints and their eligibility for conformance testing was discussed.
This issue is created to get more community input on the topic.

- [Discussion starts around 14mins](https://www.youtube.com/watch?v=31Swpnpqn4Y&t=840s)
- Should node endpoints be tested?
- Can we accualy test these endpoints with the current test framework
- [Conformance meeting notes](https://docs.google.com/document/d/1W31nXh9RYAb_VaYkwuPLd1hFxuRX3iU0DmaQ4lkCsX8/edit#heading=h.mne7b79n77pi)

/sig testing
/sig architecture
/area conformance

### PR DESCRIPTION
**Pull request by riaankleinhans**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it:**
Add Node endpoints to list of ineligible endpoints. Was discussed in #109379 and in [SIG Node mailing group](https://groups.google.com/u/0/g/kubernetes-sig-node/c/5c00B0_ARiA).
- readCoreV1NodeStatus
- replaceCoreV1NodeStatus
- deleteCoreV1CollectionNode

**Special notes for your reviewer:**

**Which issue(s) this PR fixes:**
Fixes #109379

**Does this PR introduce a user-facing change?:**
```
NONE

```

**Release note:**
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:**
```
NONE

```

/sig architecture
/area conformance
/sig node

### CODE REVIEW

**Comment by dims** [comment]
cc @liggitt

**Review by endocrimes — COMMENTED** [review]
/lgtm

These are reasonably k/k kubelet specific rather than meaningful public api. I wouldn't expect them to be exposed in all third-party implementations or even necessarily hosted products.

File: test/conformance/testdata/ineligible_endpoints.yaml [hunk 1]
```diff
@@ -433,3 +433,13 @@
 - endpoint: replaceNetworkingV1NamespacedNetworkPolicyStatus
   reason: endpoints is currently feature gated and and will only receive e2e & conformance test in 1.25
   link: https://github.com/kubernetes/kubernetes/pull/107963
+- endpoint: readCoreV1NodeStatus
+  reason: Kubernetes distribution would reasonably not allow this actions via the API
```
Comment (smarterclayton) [inline test/conformance/testdata/ineligible_endpoints.yaml:437]: minor typo: `this action`

**Comment by smarterclayton** [comment]
One tiny review nit, but otherwise will approve once that's changed.

**Comment by smarterclayton** [comment]
/approve
/lgtm

thanks

**Pull request merged**